opsmngr/agent_api_keys.go

```diff
@@ -23,26 +23,24 @@ type AgentAPIKeysServiceOp struct {
 }
 
 type AgentAPIKey struct {
-	ID            string  `json:"id"`
+	ID            string  `json:"_id"`
 	Key           string  `json:"key"`
 	Desc          string  `json:"desc"`
-	CreatedTime   string  `json:"createdTime"`
-	CreatedUserID *string `json:"createdUserId"`
-	CreatedIPAddr *string `json:"createdIpAddr"`
+	CreatedTime   int64   `json:"createdTime"`
+	CreatedUserID *string `json:"createdUserId,omitempty"`
+	CreatedIPAddr *string `json:"createdIpAddr,omitempty"`
 	CreatedBy     string  `json:"createdBy"`
 }
 
-type AgentAPIKeys struct {
-	Links      []*atlas.Link  `json:"links"`
-	Results    []*AgentAPIKey `json:"results"`
-	TotalCount int            `json:"totalCount"`
+type AgentAPIKeysRequest struct {
+	Desc string `json:"desc"`
 }
 
 // See more: https://docs.opsmanager.mongodb.com/current/reference/api/agentapikeys/create-one-agent-api-key/
-func (s *AgentAPIKeysServiceOp) Create(ctx context.Context, projectID string) (*AgentAPIKey, *atlas.Response, error) {
+func (s *AgentAPIKeysServiceOp) Create(ctx context.Context, projectID string, agent AgentAPIKeysRequest) (*AgentAPIKey, *atlas.Response, error) {
```

Comment:
already said this on the interface but don't forget to change this to a pointer as well